src/Tasks/Microsoft.NET.Build.Tasks/CheckForDuplicateItems.cs

```diff
@@ -27,11 +27,18 @@ public class CheckForDuplicateItems : TaskBase
         [Required]
         public string MoreInformationLink { get; set; }
 
+        [Output]
+        public ITaskItem [] DeduplicatedItems { get; set; }
+
         protected override void ExecuteCore()
         {
+            DeduplicatedItems = Array.Empty<ITaskItem>();
+
             if (DefaultItemsEnabled && DefaultItemsOfThisTypeEnabled)
             {
-                var duplicateItems = Items.GroupBy(i => i.ItemSpec).Where(g => g.Count() > 1).ToList();
+                var itemGroups = Items.GroupBy(i => i.ItemSpec);
```

Comment:
I would think it would be the opposite: e.g. case-insensitive better for Windows, insensitive better for Linux. It would be valid (though asking for serious trouble) on some file systems to have two different files differing only by case and we`d insert an error where compilation would have succeeded. I am not sure what the best answer is here.